lib/core/viewmodels/schedule_settings_viewmodel.dart

```diff
@@ -80,6 +123,38 @@ class ScheduleSettingsViewModel
     _showTodayBtn =
         settings[PreferencesFlag.scheduleSettingsShowTodayBtn] as bool;
 
+    scheduleActivitiesByCourse.clear();
+    final schedulesActivities = await _courseRepository.getScheduleActivities();
+    for (final activity in schedulesActivities) {
+      if (activity.activityCode == ActivityCode.labGroupA ||
+          activity.activityCode == ActivityCode.labGroupB) {
+        // Create the list with the new activity inside or add the activity to an existing group
+        if (!_scheduleActivitiesByCourse.containsKey(activity.courseAcronym)) {
+          _scheduleActivitiesByCourse[activity.courseAcronym] = [activity];
+        } else {
+          // Remove duplicata
```

Comment:
```suggestion
          // Add the activity to the course.
```
I think this is what you meant